extensions/frontend-tiles/src/GraphicsProvider/GraphicRepresentationProvider.ts

```diff
@@ -0,0 +1,263 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, Logger} from "@itwin/core-bentley";
+import { loggerCategory} from "../LoggerCategory";
+
+/** The expected format of the Graphic Representation
+ * @beta
+ */
+/* eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents */
+export type GraphicRepresentationFormat = "IMDL" | "3DTILES" | string;
+
+/** Graphic representations are generated from Data Sources.
+ * The status of a Graphic Representation indicates the progress of that generation process.
+ * @beta
+ */
+export enum GraphicRepresentationStatus {
+  InProgress = "In progress",
+  Complete = "Complete",
+  NotStarted = "Not started",
+  Failed = "Failed",
+}
+
+/**
+ * Represents a data source for a graphic representation.
+ * A data source is usually higher-fidelity and contains more information, but may not be as well suited for visualization
+ * as a graphic representation.
+ * @beta
+ */
+export interface DataSource {
+  /** The iTwinId associated with the DataSource */
+  iTwinId: string;
+  /** The unique identifier of a DataSource.
+   * For example, a DataSource of type "IMODEL" has an iModelId which would be attributed to this value.
+   */
+  id: string;
+  /** The unique identifier for a specific version of a DataSource.
+   * For example, if a specific version of an iModel is desired, the iModel's changesetId would be attributed to this value.
+  */
+  changeId?: string;
+  /** The type of the data source. For example, a DataSource can be of type "IMODEL" or "RealityData" */
+  type: string;
+}
+
+/** Represents a visual representation of a data source, for example a 3d tileset.
+ * A data source can be an iModel, reality data, or other kind of graphical data.
+ * @see [[queryGraphicRepresentations]] for its construction as a representation of the data produced by a query of data sources.
+ * @beta
+ */
+export interface GraphicRepresentation {
+  /** The display name of the Graphic Representation */
+  displayName: string;
+  /** The unique identifier for the Graphic Representation */
+  representationId: string;
+  /** The status of the generation of the Graphic Representation from its Data Source.
+   * @see [[GraphicRepresentationStatus]] for possible values.
+   */
+  status: GraphicRepresentationStatus;
+  /** The expected format of the Graphic Representation
+   * @see [[GraphicRepresentationFormat]] for possible values.
+   */
+  format:  GraphicRepresentationFormat;
+  /** The url of the Graphic Representation */
+  url?: string;
```

Comment:
Under what circumstances would `url` be undefined?

Reply:
If the status of the graphic representation is not "complete", it won't be guaranteed to have a url for the data. For example if you query a representation that was recently created it might still be "in progress" or if there was an error during generation it could be "failed" and a url may not have been added yet. Do you think it makes sense for it to be optional if this is explained in the docs?

Reply:
Will `url` be valid if and only if `status` === "Complete"?

Reply:
`url` is not valid if and only if `status` === "Complete", but that is the only time we can guarantee that it is valid. Currently, the url will show up at some point while the graphic representation is being generated, meaning at some point while the `status` === "In Progress". Until that point the url is undefined. But this means if you query a representation and get one that is "In Progress", there's a chance there is a url, but we can't guarantee it, and it might be undefined. @eringram correct me if I'm wrong on any of this or if you want to add anything.

Reply:
This change has been pushed, GraphicRepresentation is now a type with the above structure.

Reply:
I had no idea that was possible in typescript, amazing, thanks @pmconne !!